test/testPartyAttachmentsAfterUpdate.js

```diff
@@ -0,0 +1,659 @@
+            const attachmentKey3b = uuid.v4();
+            const attachmentKey4b = uuid.v4();
+
+            const response = await httpClient.put({ path: resourceHref, body });
+            assert.equal(response.status, 201);
+
+            debug("PUTting the profile images as attachments");
+            const filesToPut = [
+              {
+                remotefileName: "profile1 *% .png",
+                localFilename: "test/images/orange-boy-icon.png",
+                attachmentKey: attachmentKey1,
+                resourceHref,
+              },
+              {
+                remotefileName: "profile2.png",
+                localFilename: "test/images/little-boy-white.png",
+                attachmentKey: attachmentKey2,
+                resourceHref,
+              },
+              {
+                remotefileName: "profile3.png",
+                localFilename: "test/images/avatar-black.png",
+                attachmentKey: attachmentKey3,
+                resourceHref,
+              },
+              {
+                remotefileName: "profile4.png",
+                localFilename: "test/images/avatar-blue.png",
+                attachmentKey: attachmentKey4,
+                resourceHref,
+              },
+            ];
+
+            clearCheckStoreAttachmentsReceivedList(checkStoreAttachmentsReceivedList);
+            await uploadFilesAndCheck(httpClient, filesToPut);
+            verifyCheckStoreAttachmentsReceivedListOrder(checkStoreAttachmentsReceivedList);
+
+            // Multiple upload with one extra attachment which is a copy of an existing attachment
+            const filesToPut2 = [
+              {
+                remotefileName: "profile1 *% b.png",
+                urlToCopy: attachmentDownloadUrl1,
+                attachmentKey: attachmentKey1b,
+                resourceHref,
+              },
+              {
+                remotefileName: "profile2b.png",
+                urlToCopy: attachmentDownloadUrl2,
+                attachmentKey: attachmentKey2b,
+                resourceHref,
+              },
+              {
+                remotefileName: "profile3b.png",
+                urlToCopy: attachmentDownloadUrl3,
+                attachmentKey: attachmentKey3b,
+                resourceHref,
+              },
+              {
+                remotefileName: "profile4b.png",
+                urlToCopy: attachmentDownloadUrl4,
+                attachmentKey: attachmentKey4b,
+                resourceHref,
+              },
+            ];
+
+            clearCheckStoreAttachmentsReceivedList(checkStoreAttachmentsReceivedList);
+            await uploadFilesAndCheck(httpClient, filesToPut2);
+            verifyCheckStoreAttachmentsReceivedListOrder(checkStoreAttachmentsReceivedList);
+
+          } catch (err) {
+            // console.log(err.err)
+            assert.fail(err.err);
+          }
+        });
+
+        it("UPLOAD and COPY via /copy", async () => {
```

Comment:
Only a few lines are different here compared with the previous test, how about sharing the code in a before hook and the describe body?

Reply:
might be future improvement